### PR TITLE
Added hmmer/hmmfetch module

### PR DESCRIPTION
**Pull request by erikrikarddaniel**
## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

This adds the hmmfetch command from the HMMER suite.

The tool can be used in three different ways: a key on the command line, a file with keys or without either and the `--index` flag to produce an index for later use.

### CODE REVIEW

File: modules/nf-core/hmmer/hmmfetch/main.nf [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    conda "bioconda::hmmer=3.3.2"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/hmmer:3.3.2--h87f3376_2':
+        'biocontainers/hmmer:3.3.2--h87f3376_2' }"
+
+    // The module can be called with either a key, a file containing keys or neither.
+    // In the latter case, the hmm database will be indexed and an index but no output
+    // hmm will be produced.
+    input:
+    path  hmm
```
[the remark this answers is not in the document]
Comment (erikrikarddaniel) [reply modules/nf-core/hmmer/hmmfetch/main.nf:14]: I saw no point in having a meta, but why not? I'll fix that.

[the remark this answers is not in the document]
Comment (nvnieuwk) [reply modules/nf-core/hmmer/hmmfetch/main.nf:14]: Allright, I think it's better to add too much meta than not enough meta :p

[the remark this answers is not in the document]
Comment (erikrikarddaniel) [reply modules/nf-core/hmmer/hmmfetch/main.nf]: For the index case, I can't see any reason to have a meta. It's just a file of the same name as the database with a different suffix. It's not mentioned when calling the tool, just present in the same directory. The param is just there to stage the index file.

[the remark this answers is not in the document]
Comment (nvnieuwk) [reply modules/nf-core/hmmer/hmmfetch/main.nf]: Okay, then I'll continue my review

File: modules/nf-core/hmmer/hmmfetch/main.nf [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    // In the latter case, the hmm database will be indexed and an index but no output
+    // hmm will be produced.
+    input:
+    tuple val(meta), path(hmm)
+    val   key
+    path  keyfile
+    path  index         // Only used to stage the index from a previous run
+
+    output:
+    path "selection.hmm", emit: hmm,   optional: true
```
Comment (nvnieuwk) [inline modules/nf-core/hmmer/hmmfetch/main.nf:20]:
```suggestion
    tuple val(meta), path("selection.hmm"), emit: hmm,   optional: true
```

File: modules/nf-core/hmmer/hmmfetch/main.nf [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    path "*.ssi"        , emit: index, optional: true
```
Comment (nvnieuwk) [inline modules/nf-core/hmmer/hmmfetch/main.nf:21]:
```suggestion
    tuple val(meta), path("*.ssi")        , emit: index, optional: true
```

File: modules/nf-core/hmmer/hmmfetch/main.nf [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args    = task.ext.args ?: ''
+    def keyarg  = key ?: ''
+    def kfopt   = keyfile ? '-f' : ''
+    def index   = ! key && ! keyfile ? '--index' : ''
+    def outfile = ! key && ! keyfile ? '' : '> selection.hmm'
```
Comment (nvnieuwk) [inline modules/nf-core/hmmer/hmmfetch/main.nf:32]: I would maybe use `prefix` here too instead of `selection` (especially since meta is now available)

Comment (erikrikarddaniel) [reply modules/nf-core/hmmer/hmmfetch/main.nf:32]: Good point.

File: modules/nf-core/hmmer/hmmfetch/main.nf [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    """
+    touch selection.hmm
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        hmmer: \$(hmmsearch -h | grep -o '^# HMMER [0-9.]*' | sed 's/^# HMMER *//')
+    END_VERSIONS
+    """
+}
```
Comment (nvnieuwk) [inline modules/nf-core/hmmer/hmmfetch/main.nf:62]: Can you also fix the stub to use prefix?

File: modules/nf-core/hmmer/hmmfetch/meta.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      pattern: "*.hmm"
+  - key:
+      type: string
+      description: Name of HMM to extract. Specify either this or keyfile. If none is specified, an index will be built.
+  - keyfile:
+      type: file
+      description: File containing list of HMM models to extract. Specify either this or key. If none is specified, an index will be built.
+      pattern: "*.txt"
+
+output:
```
Comment (nvnieuwk) [inline modules/nf-core/hmmer/hmmfetch/meta.yml:37]: Can you add meta to the output?

**Review by nvnieuwk — COMMENTED** [review]
2 more comments and then you are good to go!

File: modules/nf-core/hmmer/hmmfetch/meta.yml (hunk 6 above)
Comment (erikrikarddaniel) [reply modules/nf-core/hmmer/hmmfetch/meta.yml:37]: Thanks for reminding me!

**Review by nvnieuwk — APPROVED** [review]
LGTM!

**Comment by erikrikarddaniel** [comment]
Thanks!

**Pull request merged**